src/UnassignedAnalyser.php

```diff
@@ -29,20 +29,34 @@ public function __construct(
     public function analyse(Configuration $configuration): array
     {
         $astMap = $this->astRunner->createAstMapByFiles($this->fileResolver->resolve($configuration), $configuration->getAnalyser());
-        $classLikeLayerResolver = $this->classLikeLayerResolverFactory->create($configuration, $astMap);
+        $tokenLayerResolver = $this->tokenLayerResolverFactory->create($configuration, $astMap);
 
-        /** @var string[] $classLikeNames */
-        $classLikeNames = [];
+        /** @var string[] $tokenNames */
+        $tokenNames = [];
 
-        foreach ($astMap->getAstClassReferences() as $classReference) {
-            $classLikeName = $classReference->getTokenName();
-            if ([] === $classLikeLayerResolver->getLayersByTokenName($classLikeName)) {
-                $classLikeNames[] = $classLikeName->toString();
+        foreach ($astMap->getAstFileReferences() as $fileReference) {
```

Comment:
This is an extension of the current functionality. But I am thinking if this should first pass by Emitters somehow since now you might not want to collect some of the tokens. Something like to actually analyze normally, and look at the `dependencyResult` for dependencies that are not in any layer.

Reply:
That's a good idea, but we can defer it to a separate PR.